lib/spack/spack/util/compression.py

```diff
@@ -39,6 +40,39 @@ def _gunzip(archive_file):
             f_out.write(f_in.read())
 
 
+def spack_unzipper():
+    from zipfile import ZipFile, ZipInfo
+
+    # implementation based on https://stackoverflow.com/a/39296577
+    class SpackZipFile(ZipFile):
+        if sys.version_info < (3, 6):
+            def extract(self, member, path=None, pwd=None):
+                # below is a direct copy of ZipFile's extract method
```

Comment:
I don't understand: if this is an exact copy: why is it redefined here? Why not:

```
if sys.version_info >= (3, 7):
  def _extract_member(...):
    ...
# no else statement needed
```

Reply:
That comment is a mistake that I left from a first pass and missed on cleanup. I do agree that we only need to override `_extract_member`, but I was confused as to the API changes per version, I just went through the source and it looks like as long as we override `_extract_member` the version will not matter.